src/RadhydroPulseMGint/test_radhydro_pulse_MG_int.cpp

```diff
@@ -158,31 +165,53 @@ auto compute_kappa(const double nu, const double Tgas) -> double
 	// cm^-1
 	auto T_ = Tgas / T_ref;
 	auto nu_ = nu / nu_ref;
-	return kappa0 * std::pow(T_, -0.5) * std::pow(nu_, -3) * (1.0 - std::exp(-coeff_ * nu_ / T_));
+	return scaleup * kappa0 * std::pow(T_, -0.5) * std::pow(nu_, -3) * (1.0 - std::exp(-coeff_ * nu_ / T_));
 }
 
 template <>
 AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE auto
 RadSystem<MGProblem>::DefineOpacityExponentsAndLowerValues(amrex::GpuArray<double, nGroups_ + 1> const rad_boundaries, const double rho,
-							   const double Tgas) -> amrex::GpuArray<amrex::GpuArray<double, nGroups_>, 2>
+							   const double Tgas) -> amrex::GpuArray<amrex::GpuArray<double, nGroups_ + 1>, 2>
 {
-	amrex::GpuArray<double, nGroups_> exponents{};
-	amrex::GpuArray<double, nGroups_> kappa_lower{};
-	for (int g = 0; g < nGroups_; ++g) {
-		auto kappa_up = compute_kappa(rad_boundaries[g + 1], Tgas);
-		auto kappa_down = compute_kappa(rad_boundaries[g], Tgas);
-		exponents[g] = std::log(kappa_up / kappa_down) / std::log(rad_boundaries[g + 1] / rad_boundaries[g]);
-		kappa_lower[g] = kappa_down / rho;
+	amrex::GpuArray<double, nGroups_ + 1> exponents{};
+	amrex::GpuArray<double, nGroups_ + 1> kappa_lower{};
+
+	for (int g = 0; g < nGroups_ + 1; ++g) {
+		if (RadSystem_Traits<MGProblem>::opacity_model == OpacityModel::piecewise_constant_opacity) {
+			exponents[g] = 0.0;
+			if (g < n_groups_) {
+				auto nu_center = std::sqrt(rad_boundaries[g] * rad_boundaries[g + 1]);
+				kappa_lower[g] = compute_kappa(nu_center, Tgas) / rho;
+			}
+			// note kappa_lower[nGroups_] is not used
+		} else {
+			if (g == n_groups_) {
+				exponents[g] = 0.;
+				kappa_lower[g] = compute_kappa(rad_boundaries[g], Tgas) / rho;
+			} else {
+				auto kappa_up = compute_kappa(rad_boundaries[g + 1], Tgas);
+				auto kappa_down = compute_kappa(rad_boundaries[g], Tgas);
+				exponents[g] = std::log(kappa_up / kappa_down) / std::log(rad_boundaries[g + 1] / rad_boundaries[g]);
+				kappa_lower[g] = kappa_down / rho;
+			}
+		}
 		AMREX_ASSERT(!std::isnan(exponents[g]));
 		AMREX_ASSERT(kappa_lower[g] >= 0.);
 	}
-	amrex::GpuArray<amrex::GpuArray<double, nGroups_>, 2> const exponents_and_values{exponents, kappa_lower};
+
+	// dummy test
+	// for (int g = 0; g < nGroups_ + 1; ++g) {
+	// 	exponents[g] = 0.0;
+	// 	kappa_lower[g] = kappa0;
+	// }
+
```

Comment:
Remove commented-out testing code for cleanliness / ease of reading.

Reply:
Done.